src/qtgui/meter.cpp

```diff
@@ -83,167 +69,94 @@ QSize CMeter::sizeHint() const
     return QSize(100, 30);
 }
 
-void CMeter::resizeEvent(QResizeEvent *)
-{
-    if (!size().isValid())
-        return;
-
-    if (m_Size != size())
-    {
-        // if size changed, resize pixmaps to new screensize
-        m_Size = size();
-        qreal dpr = devicePixelRatioF();
-        m_OverlayPixmap = QPixmap(m_Size.width() * dpr, m_Size.height() * dpr);
-        m_OverlayPixmap.setDevicePixelRatio(dpr);
-        m_OverlayPixmap.fill(Qt::black);
-        m_2DPixmap = QPixmap(m_Size.width() * dpr, m_Size.height() * dpr);
-        m_2DPixmap.setDevicePixelRatio(dpr);
-        m_2DPixmap.fill(Qt::black);
-
-        qreal w = (m_2DPixmap.width() / dpr) - 2 * CTRL_MARGIN * (m_2DPixmap.width() / dpr);
-        m_pixperdb = w / fabs((double)(MAX_DB - MIN_DB));
-        setSqlLevel(m_Sql);
-    }
-
-    DrawOverlay();
-    draw();
-}
-
 void CMeter::setLevel(float dbfs)
 {
-    if (dbfs < MIN_DB)
-        dbfs = MIN_DB;
-    else if (dbfs > MAX_DB)
-        dbfs = MAX_DB;
-
-    float level = m_dBFS;
-    float alpha  = dbfs < level ? ALPHA_DECAY : ALPHA_RISE;
-    m_dBFS -= alpha * (level - dbfs);
-    m_Siglevel = (qreal)(level - MIN_DB) * m_pixperdb;
```

Comment:
The meter should also be slightly more responsive now that it's drawing the current signal level instead of the previous one from 100ms ago. :smile: